tests/Integration/Behaviour/Features/Context/Domain/OrderPaymentFeatureContext.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PrestaShop\PrestaShop\Core\Domain\Order\Payment\Command\AddPaymentCommand;
+use PrestaShop\PrestaShop\Core\Domain\Order\Query\GetOrderForViewing;
+use PrestaShop\PrestaShop\Core\Domain\Order\QueryResult\OrderForViewing;
+use PrestaShop\PrestaShop\Core\Domain\Order\QueryResult\OrderPaymentForViewing;
+use PrestaShop\PrestaShop\Core\Domain\Order\QueryResult\OrderPaymentsForViewing;
+use PrestaShopException;
+use RuntimeException;
+use Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+class OrderPaymentFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I add payment to order with id :orderId with the following properties:
+     *
+     * @param int $orderId
+     * @param TableNode $table
+     *
+     * @throws RuntimeException
+     * @throws ServiceCircularReferenceException
+     * @throws ServiceNotFoundException
+     */
+    public function iAddPaymentToWithIdOrderWithIdTheFollowingProperties(int $orderId, TableNode $table)
+    {
+        /** @var array $hash */
+        $hash = $table->getHash();
+        if (count($hash) != 1) {
+            throw new RuntimeException('Payment details are invalid');
+        }
+        /** @var array $data */
+        $data = $hash[0];
+
+        $this->getCommandBus()->handle(
+            new AddPaymentCommand(
+                $orderId,
+                $data['date'],
+                $data['payment_method'],
+                $data['amount'],
+                (int) $data['id_currency'],
+                $data['id_invoice'],
+                $data['transaction_id']
+            )
+        );
+    }
+
+    /**
+     * @Then if I query order with id :orderId payments I should get :numberOfPayments payments
```

Comment:
```suggestion
     * @Then order :orderId should have :numberOfPayments payments
```
Make it short, make it clear, make it readable 😉